pkg/matcher/annotation_matcher.go

```diff
@@ -26,7 +26,7 @@ const (
 
 func branchMatch(prunBranch, baseBranch string) bool {
 	// if target is refs/heads/.. and base is without ref (for pullRequest)
-	if strings.HasPrefix(prunBranch, "refs/heads") && !strings.Contains(baseBranch, "/") {
+	if strings.HasPrefix(prunBranch, "refs/heads") && !strings.Contains(baseBranch, "refs/heads") {
```

Comment:
i dont think this if is required, we can simply do

```
if strings.HasPrefix(prunBranch, "refs/heads/")  {
		ref := "refs/heads/" + prunBranch
	}
```
similarly for baseBranch and later glob compile